### PR TITLE
Adding SSR support

### LINKED ISSUE #1: Problems installing new version
Hello and thanks for the new version.

Unfortunately, when uninstalling the old version and installing the new one, I am having a problem compiling my project with Gatsby.

```
success Writing page-data.json files to public directory - 0.010s - 7/7 705.39/s
success Building HTML renderer - 2.764s
failed Building static HTML for pages - 0.654s

 ERROR #95312 

"document" is not available during server side rendering.

See our docs page for more info on this error: https://gatsby.dev/debug-html


  19637 |   var key = options.key;
  19638 |   if (key === "css") {
> 19639 |     var ssrStyles = document.querySelectorAll("style[data-emotion]:not([data-s])");
        | ^
  19640 |     Array.prototype.forEach.call(ssrStyles, function(node2) {
  19641 |       var dataEmotionAttribute = node2.getAttribute("data-emotion");
  19642 |       if (dataEmotionAttribute.indexOf(" ") === -1) {


  WebpackError: ReferenceError: document is not defined


  - index.es.js:19639 
    [gatsby-starter-default]/[react-dark-mode-toggle-2]/dist/index.es.js:19639:1
  
  - index.es.js:19978 
    [gatsby-starter-default]/[react-dark-mode-toggle-2]/dist/index.es.js:19978:1
  
  - index.es.js:20072 
```

https://www.gatsbyjs.com/docs/debugging-html-builds/

Greetings

### LINKED ISSUE #6: SSR Not Supported
Hi!

We tried to use the toggle in our Next.JS project, but We got an "Document not defined" error. This has happened duo to the emotion package, which using Document.

I have forked the package and added a fix to the issue, By removing emotion, and using css only.

I hope that PR will help that package to grow and be more useful for those preferring SSR!

### PR DESCRIPTION
**Pull request by orinaveh**
#6

### CODE REVIEW

**Comment by orinaveh** [comment]
@todd-elvers Can you review my PR?

**Comment by trevorlasn** [comment]
@todd-elvers

File: src/index.module.css
Comment (todd-elvers) [inline src/index.module.css:1]: My only concern with this approach is vendor prefixing.  Unless I missing something and `vite-plugin-css-injected-by-js` handles it for you, what about changing it to the following:

```
.button {
      cursor: pointer;
      overflow: hidden;
      -webkit-appearance: none;
         -moz-appearance: none;
              appearance: none;
      border: none;
      background-color: transparent;
      padding: 0;
}

.player {
      display: none;
      -webkit-box-align: center;
         -ms-flex-align: center;
            align-items: center;
      -webkit-box-pack: center;
         -ms-flex-pack: center;
       justify-content: center;
}

.player--loaded {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
}
```

**Review by todd-elvers — COMMENTED** [review]
Everything looks good!  Just posed one question regarding vendor prefixing.  If `vite-plugin-css-injected-by-js` handles vendor prefixing just let me know and we can ignore that comment.

**Pull request merged**